### ISSUE TITLE
No available features on Ubuntu 22.04

### ISSUE THREAD

**Issue by Baa14453** [f4exb/sdrangel#1310]
I compiled SDRAngel and when I try to add a feature I cannot open the drop-down, unsure if the drop-down menu is not working or if there are no features available.

![image](https://user-images.githubusercontent.com/9057997/175776763-c440f3d0-35cf-46a4-85d4-ec87ac2de873.png)

Other drop-down menus work fine:

![image](https://user-images.githubusercontent.com/9057997/175776780-e36fa4db-08b4-435c-bcb4-c4864fc1ea6d.png)

I followed the compile guide []here(https://github.com/f4exb/sdrangel/wiki/Compile-from-source-in-Linux) with no hardware dependencies added.

I'm using Ubuntu 22.04 with Wayland.

Log:
```
2022-06-25 16:05:02.279 (I) Warning: Ignoring XDG_SESSION_TYPE=wayland on Gnome. Use QT_QPA_PLATFORM=wayland to run on Wayland anyway.
2022-06-25 16:05:02.427 (I) SDRangel 7.3.2-60-g77963f9eb Qt 5.15.3 64b x86_64 Ubuntu 22.04 LTS DSP Rx:24b Tx:16b PID 1993
2022-06-25 16:05:02.432 (I) MainSettings::MainSettings: settings file: format: 0 location: /home/ubuntu-2204/.config/f4exb/SDRangel.conf
2022-06-25 16:05:02.432 (D) MessagePipes::startGC
2022-06-25 16:05:02.432 (D) DataPipes::startGC
2022-06-25 16:05:02.432 (D) AudioDeviceManager::AudioDeviceManager: scan input devices
2022-06-25 16:05:02.519 (D) AudioDeviceManager::AudioDeviceManager: input device #0: default
2022-06-25 16:05:02.519 (D) AudioDeviceManager::AudioDeviceManager: input device #1: sysdefault
2022-06-25 16:05:02.519 (D) AudioDeviceManager::AudioDeviceManager: input device #2: iec958
2022-06-25 16:05:02.519 (D) AudioDeviceManager::AudioDeviceManager: input device #3: samplerate
2022-06-25 16:05:02.519 (D) AudioDeviceManager::AudioDeviceManager: input device #4: speexrate
2022-06-25 16:05:02.519 (D) AudioDeviceManager::AudioDeviceManager: input device #5: jack
2022-06-25 16:05:02.519 (D) AudioDeviceManager::AudioDeviceManager: input device #6: oss
2022-06-25 16:05:02.519 (D) AudioDeviceManager::AudioDeviceManager: input device #7: pulse
2022-06-25 16:05:02.519 (D) AudioDeviceManager::AudioDeviceManager: input device #8: upmix
2022-06-25 16:05:02.519 (D) AudioDeviceManager::AudioDeviceManager: input device #9: vdownmix
2022-06-25 16:05:02.519 (D) AudioDeviceManager::AudioDeviceManager: input device #10: hw:CARD=I82801AAICH,DEV=0
2022-06-25 16:05:02.519 (D) AudioDeviceManager::AudioDeviceManager: input device #11: hw:CARD=I82801AAICH,DEV=1
2022-06-25 16:05:02.519 (D) AudioDeviceManager::AudioDeviceManager: input device #12: plughw:CARD=I82801AAICH,DEV=0
2022-06-25 16:05:02.519 (D) AudioDeviceManager::AudioDeviceManager: input device #13: plughw:CARD=I82801AAICH,DEV=1
2022-06-25 16:05:02.519 (D) AudioDeviceManager::AudioDeviceManager: input device #14: sysdefault:CARD=I82801AAICH
2022-06-25 16:05:02.519 (D) AudioDeviceManager::AudioDeviceManager: input device #15: front:CARD=I82801AAICH,DEV=0
2022-06-25 16:05:02.519 (D) AudioDeviceManager::AudioDeviceManager: input device #16: dsnoop:CARD=I82801AAICH,DEV=0
2022-06-25 16:05:02.519 (D) AudioDeviceManager::AudioDeviceManager: input device #17: dsnoop:CARD=I82801AAICH,DEV=1
2022-06-25 16:05:02.519 (D) AudioDeviceManager::AudioDeviceManager: input device #18: usbstream:CARD=I82801AAICH
2022-06-25 16:05:02.519 (D) AudioDeviceManager::AudioDeviceManager: input device #19: alsa_input.pci-0000_00_05.0.analog-stereo
2022-06-25 16:05:02.519 (D) AudioDeviceManager::AudioDeviceManager: input device #20: alsa_output.pci-0000_00_05.0.analog-stereo.monitor
2022-06-25 16:05:02.519 (D) AudioDeviceManager::AudioDeviceManager: scan output devices
2022-06-25 16:05:02.615 (D) AudioDeviceManager::AudioDeviceManager: output device #0: default
2022-06-25 16:05:02.616 (D) AudioDeviceManager::AudioDeviceManager: output device #1: sysdefault
2022-06-25 16:05:02.616 (D) AudioDeviceManager::AudioDeviceManager: output device #2: iec958
2022-06-25 16:05:02.616 (D) AudioDeviceManager::AudioDeviceManager: output device #3: samplerate
2022-06-25 16:05:02.616 (D) AudioDeviceManager::AudioDeviceManager: output device #4: speexrate
2022-06-25 16:05:02.616 (D) AudioDeviceManager::AudioDeviceManager: output device #5: jack
2022-06-25 16:05:02.616 (D) AudioDeviceManager::AudioDeviceManager: output device #6: oss
2022-06-25 16:05:02.616 (D) AudioDeviceManager::AudioDeviceManager: output device #7: pulse
2022-06-25 16:05:02.616 (D) AudioDeviceManager::AudioDeviceManager: output device #8: upmix
2022-06-25 16:05:02.616 (D) AudioDeviceManager::AudioDeviceManager: output device #9: vdownmix
2022-06-25 16:05:02.616 (D) AudioDeviceManager::AudioDeviceManager: output device #10: hw:CARD=I82801AAICH,DEV=0
2022-06-25 16:05:02.616 (D) AudioDeviceManager::AudioDeviceManager: output device #11: plughw:CARD=I82801AAICH,DEV=0
2022-06-25 16:05:02.616 (D) AudioDeviceManager::AudioDeviceManager: output device #12: sysdefault:CARD=I82801AAICH
2022-06-25 16:05:02.616 (D) AudioDeviceManager::AudioDeviceManager: output device #13: front:CARD=I82801AAICH,DEV=0
2022-06-25 16:05:02.616 (D) AudioDeviceManager::AudioDeviceManager: output device #14: surround21:CARD=I82801AAICH,DEV=0
2022-06-25 16:05:02.616 (D) AudioDeviceManager::AudioDeviceManager: output device #15: surround40:CARD=I82801AAICH,DEV=0
2022-06-25 16:05:02.616 (D) AudioDeviceManager::AudioDeviceManager: output device #16: surround41:CARD=I82801AAICH,DEV=0
2022-06-25 16:05:02.616 (D) AudioDeviceManager::AudioDeviceManager: output device #17: surround50:CARD=I82801AAICH,DEV=0
2022-06-25 16:05:02.616 (D) AudioDeviceManager::AudioDeviceManager: output device #18: surround51:CARD=I82801AAICH,DEV=0
2022-06-25 16:05:02.616 (D) AudioDeviceManager::AudioDeviceManager: output device #19: iec958:CARD=I82801AAICH,DEV=0
2022-06-25 16:05:02.616 (D) AudioDeviceManager::AudioDeviceManager: output device #20: dmix:CARD=I82801AAICH,DEV=0
2022-06-25 16:05:02.616 (D) AudioDeviceManager::AudioDeviceManager: output device #21: usbstream:CARD=I82801AAICH
2022-06-25 16:05:02.616 (D) AudioDeviceManager::AudioDeviceManager: output device #22: alsa_output.pci-0000_00_05.0.analog-stereo
2022-06-25 16:05:02.618 (D) MainWindow::MainWindow: start
2022-06-25 16:05:02.675 (D) FFTEngine::create: using FFTW engine
2022-06-25 16:05:02.676 (D) FFTWEngine::configure: no FFTW wisdom file
2022-06-25 16:05:02.760 (D) FFT: creating FFTW plan (n=128,forward) took 83 ms
2022-06-25 16:05:02.760 (D) FFTEngine::create: using FFTW engine
2022-06-25 16:05:02.760 (D) FFTWEngine::configure: no FFTW wisdom file
2022-06-25 16:05:02.923 (D) FFT: creating FFTW plan (n=256,forward) took 162 ms
2022-06-25 16:05:02.923 (D) FFTEngine::create: using FFTW engine
2022-06-25 16:05:02.923 (D) FFTWEngine::configure: no FFTW wisdom file
2022-06-25 16:05:03.210 (D) FFT: creating FFTW plan (n=512,forward) took 287 ms
2022-06-25 16:05:03.210 (D) FFTEngine::create: using FFTW engine
2022-06-25 16:05:03.210 (D) FFTWEngine::configure: no FFTW wisdom file
2022-06-25 16:05:03.657 (D) FFT: creating FFTW plan (n=1024,forward) took 446 ms
2022-06-25 16:05:03.660 (D) MainWindow::MainWindow: load settings...
2022-06-25 16:05:03.660 (D) MainWindow::loadSettings
2022-06-25 16:05:03.661 (D) AudioDeviceManager::deserialize
2022-06-25 16:05:03.661 (D) MainWindow::MainWindow: load plugins...
2022-06-25 16:05:03.662 (D) PluginManager::loadPluginsDir: fileName: libinputaudio.so
2022-06-25 16:05:03.665 (I) PluginManager::loadPluginsDir: loaded plugin libinputaudio.so
2022-06-25 16:05:03.665 (D) PluginManager::loadPluginsDir: fileName: libinputfcdpro.so
2022-06-25 16:05:03.674 (I) PluginManager::loadPluginsDir: loaded plugin libinputfcdpro.so
2022-06-25 16:05:03.674 (D) PluginManager::loadPluginsDir: fileName: libinputfcdproplus.so
2022-06-25 16:05:03.677 (I) PluginManager::loadPluginsDir: loaded plugin libinputfcdproplus.so
2022-06-25 16:05:03.677 (D) PluginManager::loadPluginsDir: fileName: libinputfileinput.so
2022-06-25 16:05:03.680 (I) PluginManager::loadPluginsDir: loaded plugin libinputfileinput.so
2022-06-25 16:05:03.680 (D) PluginManager::loadPluginsDir: fileName: libinputkiwisdr.so
2022-06-25 16:05:03.682 (I) PluginManager::loadPluginsDir: loaded plugin libinputkiwisdr.so
2022-06-25 16:05:03.682 (D) PluginManager::loadPluginsDir: fileName: libinputlocal.so
2022-06-25 16:05:03.686 (I) PluginManager::loadPluginsDir: loaded plugin libinputlocal.so
2022-06-25 16:05:03.686 (D) PluginManager::loadPluginsDir: fileName: libinputremote.so
2022-06-25 16:05:03.692 (I) PluginManager::loadPluginsDir: loaded plugin libinputremote.so
2022-06-25 16:05:03.692 (D) PluginManager::loadPluginsDir: fileName: libinputsigmffileinput.so
2022-06-25 16:05:03.698 (I) PluginManager::loadPluginsDir: loaded plugin libinputsigmffileinput.so
2022-06-25 16:05:03.698 (D) PluginManager::loadPluginsDir: fileName: libinputtestsource.so
2022-06-25 16:05:03.701 (I) PluginManager::loadPluginsDir: loaded plugin libinputtestsource.so
2022-06-25 16:05:03.701 (D) PluginManager::loadPluginsDir: fileName: libmimometismiso.so
2022-06-25 16:05:03.704 (I) PluginManager::loadPluginsDir: loaded plugin libmimometismiso.so
2022-06-25 16:05:03.704 (D) PluginManager::loadPluginsDir: fileName: libmimotestmi.so
2022-06-25 16:05:03.707 (I) PluginManager::loadPluginsDir: loaded plugin libmimotestmi.so
2022-06-25 16:05:03.707 (D) PluginManager::loadPluginsDir: fileName: liboutputaudio.so
2022-06-25 16:05:03.709 (I) PluginManager::loadPluginsDir: loaded plugin liboutputaudio.so
2022-06-25 16:05:03.709 (D) PluginManager::loadPluginsDir: fileName: liboutputfileoutput.so
2022-06-25 16:05:03.712 (I) PluginManager::loadPluginsDir: loaded plugin liboutputfileoutput.so
2022-06-25 16:05:03.712 (D) PluginManager::loadPluginsDir: fileName: liboutputlocal.so
2022-06-25 16:05:03.715 (I) PluginManager::loadPluginsDir: loaded plugin liboutputlocal.so
2022-06-25 16:05:03.715 (D) PluginManager::loadPluginsDir: fileName: liboutputremote.so
2022-06-25 16:05:03.717 (I) PluginManager::loadPluginsDir: loaded plugin liboutputremote.so
2022-06-25 16:05:03.717 (D) PluginManager::loadPluginsDir: fileName: liboutputtestmosync.so
2022-06-25 16:05:03.720 (I) PluginManager::loadPluginsDir: loaded plugin liboutputtestmosync.so
2022-06-25 16:05:03.720 (D) PluginManager::loadPluginsDir: fileName: liboutputtestsink.so
2022-06-25 16:05:03.723 (I) PluginManager::loadPluginsDir: loaded plugin liboutputtestsink.so
2022-06-25 16:05:03.723 (D) PluginManager::registerSampleSource  Audio Input  with source name  sdrangel.samplesource.audioinput  and hardware id  "AudioInput"
2022-06-25 16:05:03.723 (D) PluginManager::registerSampleSink  Audio output  with sink name  sdrangel.samplesink.audiooutput  and hardware id  "AudioOutput"
2022-06-25 16:05:03.723 (D) PluginManager::registerSampleSource  File device input  with source name  sdrangel.samplesource.fileinput  and hardware id  "FileInput"
2022-06-25 16:05:03.723 (D) PluginManager::registerSampleSource  File device input (SigMF)  with source name  sdrangel.samplesource.sigmffileinput  and hardware id  "SigMFFileInput"
2022-06-25 16:05:03.723 (D) PluginManager::registerSampleSink  File output  with sink name  sdrangel.samplesink.fileoutput  and hardware id  "FileOutput"
2022-06-25 16:05:03.723 (D) PluginManager::registerSampleSource  FunCube Pro Input  with source name  sdrangel.samplesource.fcdpro  and hardware id  "FCDPro"
2022-06-25 16:05:03.723 (D) PluginManager::registerSampleSource  FunCube Pro+ Input  with source name  sdrangel.samplesource.fcdproplus  and hardware id  "FCDProPlus"
2022-06-25 16:05:03.723 (D) PluginManager::registerSampleSource  KiwiSDR input  with source name  sdrangel.samplesource.kiwisdrsource  and hardware id  "KiwiSDR"
2022-06-25 16:05:03.723 (D) PluginManager::registerSampleSource  Local device input  with source name  sdrangel.samplesource.localinput  and hardware id  "LocalInput"
2022-06-25 16:05:03.723 (D) PluginManager::registerSampleSink  Local device output  with sink name  sdrangel.samplesink.localoutput  and hardware id  "LocalOutput"
2022-06-25 16:05:03.723 (D) PluginManager::registerSampleMIMO  Metis MISO  with MIMO name  sdrangel.samplemimo.metismiso  and hardware id  "MetisMISO"
2022-06-25 16:05:03.723 (D) PluginManager::registerSampleSource  Remote device input  with source name  sdrangel.samplesource.remoteinput  and hardware id  "RemoteInput"
2022-06-25 16:05:03.723 (D) PluginManager::registerSampleSink  Remote device output  with sink name  sdrangel.samplesink.remoteoutput  and hardware id  "RemoteOutput"
2022-06-25 16:05:03.723 (D) PluginManager::registerSampleMIMO  Test Multiple Input  with MIMO name  sdrangel.samplemimo.testmi  and hardware id  "TestMI"
2022-06-25 16:05:03.723 (D) PluginManager::registerSampleSink  Test Sink Output  with sink name  sdrangel.samplesink.testsink  and hardware id  "TestSink"
2022-06-25 16:05:03.723 (D) PluginManager::registerSampleSource  Test Source input  with source name  sdrangel.samplesource.testsource  and hardware id  "TestSource"
2022-06-25 16:05:03.723 (D) PluginManager::registerSampleMIMO  Test Synchronous Multiple Output  with MIMO name  sdrangel.samplemimo.testmosync  and hardware id  "TestMOSync"
2022-06-25 16:05:03.725 (D) DeviceEnumerator::enumerateRxDevices: sdrangel.samplesource.audioinput
2022-06-25 16:05:03.725 (D) DeviceEnumerator::enumerateRxDevices: sdrangel.samplesource.fileinput
2022-06-25 16:05:03.725 (D) DeviceEnumerator::enumerateRxDevices: sdrangel.samplesource.sigmffileinput
2022-06-25 16:05:03.725 (D) DeviceEnumerator::enumerateRxDevices: sdrangel.samplesource.fcdpro
2022-06-25 16:05:03.726 (D) DeviceEnumerator::enumerateRxDevices: sdrangel.samplesource.fcdproplus
2022-06-25 16:05:03.726 (D) DeviceEnumerator::enumerateRxDevices: sdrangel.samplesource.kiwisdrsource
2022-06-25 16:05:03.726 (D) DeviceEnumerator::enumerateRxDevices: sdrangel.samplesource.localinput
2022-06-25 16:05:03.726 (D) DeviceEnumerator::enumerateRxDevices: sdrangel.samplesource.remoteinput
2022-06-25 16:05:03.726 (D) DeviceEnumerator::enumerateRxDevices: sdrangel.samplesource.testsource
2022-06-25 16:05:03.726 (D) DeviceEnumerator::enumerateTxDevices: sdrangel.samplesink.audiooutput
2022-06-25 16:05:03.726 (D) DeviceEnumerator::enumerateTxDevices: sdrangel.samplesink.fileoutput
2022-06-25 16:05:03.726 (D) DeviceEnumerator::enumerateTxDevices: sdrangel.samplesink.localoutput
2022-06-25 16:05:03.726 (D) DeviceEnumerator::enumerateTxDevices: sdrangel.samplesink.remoteoutput
2022-06-25 16:05:03.726 (D) DeviceEnumerator::enumerateTxDevices: sdrangel.samplesink.testsink
2022-06-25 16:05:03.726 (D) DeviceEnumerator::enumerateMIMODevices: sdrangel.samplemimo.metismiso
2022-06-25 16:05:03.730 (D) DeviceMetisScan::scan: discovery writeDatagram sent to 127.255.255.255
2022-06-25 16:05:03.730 (D) DeviceMetisScan::scan: discovery writeDatagram sent to 10.0.2.255
2022-06-25 16:05:03.730 (D) DeviceMetisScan::scan: start 0.5 second timeout loop
2022-06-25 16:05:04.205 (D) DeviceEnumerator::enumerateMIMODevices: sdrangel.samplemimo.testmi
2022-06-25 16:05:04.205 (D) DeviceEnumerator::enumerateMIMODevices: sdrangel.samplemimo.testmosync
2022-06-25 16:05:04.205 (D) DeviceEnumerator::addNonDiscoverableDevices: start
2022-06-25 16:05:04.206 (D) MainWindow::MainWindow: load current configuration...
2022-06-25 16:05:04.206 (D) MainWindow::loadConfiguration: configuration [default | no name] 1 workspace(s) - 0 device set(s) - 0 feature(s)
2022-06-25 16:05:04.241 (D) FeatureUISet::loadFeatureSetSettings: Loading preset [default | no name]
2022-06-25 16:05:04.241 (D) FeatureUISet::loadFeatureSetSettings: 0 feature(s) in preset
2022-06-25 16:05:04.241 (D) StaticFileController: docroot=:/webapi, encoding=UTF-8, maxAge=60000
2022-06-25 16:05:04.242 (D) StaticFileController: cache timeout=60000, size=1000000
2022-06-25 16:05:04.242 (D) HttpListener: Listening on port 8091
2022-06-25 16:05:04.242 (I) WebAPIServer::start: starting web API server at http://:8091
2022-06-25 16:05:04.242 (D) MainWindow::MainWindow: end
```

**Comment by Baa14453** [comment]
I've compiled this twice, once on my PC and once within a virtual machine, both re-created the problem, however with my own PC some of the features have now appeared, although I don't think I changed anything besides launching SDRAngel again from a new terminal.

Some features are still missing though, such as the Pager Demodulator, this might be unrelated:

![image](https://user-images.githubusercontent.com/9057997/175777125-eef00c93-c9d1-4e76-8494-47cf20554494.png)


In my virtual machine the issue persists so I can still test suggestions.

**Comment by srcejon** [comment]
The Pager Demod is a channel, not a Feature. So you add that from the Device window.

**Comment by Baa14453** [comment]
> The Pager Demod is a channel, not a Feature. So you add that from the Device window.

Thanks I looked everywhere for the Channels button but couldn't see it anywhere, found it just next to the device name finally:
![image](https://user-images.githubusercontent.com/9057997/175778361-0399f38f-2746-4265-9302-fdcbbc42e790.png)

**Comment by Baa14453** [comment]
Similar to before in my virtual machine where Features are not appearing, the Channels list is also empty 

![image](https://user-images.githubusercontent.com/9057997/175778412-c2c7f3b5-bfc8-4120-b607-f17c31e18e01.png)

**Comment by srcejon** [comment]
When did you get the latest source code?

Could be related to this change: https://github.com/f4exb/sdrangel/issues/1308

Your log only has device plugins loaded - no features / channels.

**Comment by Baa14453** [comment]
> When did you get the latest source code?

I cloned a couple hours ago i'll try the commit before

**Comment by Baa14453** [comment]
> When did you get the latest source code?
> 
> Could be related to this change: #1308
> 
> Your log only has device plugins loaded - no features / channels.

Just compiled commit `58328ace0d04f662dea1cb2c719573902f7e77a6` and it worked no problem, it also resolved #1311 I'll try it outside my virtual machine now and see if it still works.

**Comment by f4exb** [comment]
There is a stray comma here: https://github.com/f4exb/sdrangel/blob/master/CMakeLists.txt#L51
...and here: https://github.com/f4exb/sdrangel/blob/master/CMakeLists.txt#L82
...and here: https://github.com/f4exb/sdrangel/blob/master/CMakeLists.txt#L82
...and here: https://github.com/f4exb/sdrangel/blob/master/CMakeLists.txt#L106

This is probably the issue.
